test/controllers/TransferAgreementController2Spec.scala

```diff
@@ -0,0 +1,274 @@
+      val controller = taHelper.instantiateTransferAgreement2Controller(getAuthorisedSecurityComponents, app.configuration)
+      taHelper.mockGetConsignmentStatusGraphqlResponse(app.configuration, Some("InProgress"))
+
+      val incompleteTransferAgreementForm: Seq[(String, String)] = taHelper.getTransferAgreementForm("notCompliance", 2)
+
+      val transferAgreementSubmit = controller.transferAgreementSubmit(consignmentId)
+        .apply(FakeRequest(POST, f"/consignment/$consignmentId/transfer-agreement2")
+          .withFormUrlEncodedBody(incompleteTransferAgreementForm:_*)
+          .withCSRFToken)
+      val transferAgreementPageAsString = contentAsString(transferAgreementSubmit)
+      val pageOptions: Set[String] = incompleteTransferAgreementForm.map{
+        case (pageOption: String, _: String) => pageOption
+      }.toSet
+
+      playStatus(transferAgreementSubmit) mustBe BAD_REQUEST
+      taHelper.checkHtmlOfComplianceFormOptions.checkForOptionAndItsAttributes(transferAgreementPageAsString, incompleteTransferAgreementForm.toMap)
+      transferAgreementPageAsString must include("govuk-error-message")
+      transferAgreementPageAsString must include("error")
+      taHelper.checkHtmlContentForErrorSummary(transferAgreementPageAsString, taHelper.compliance, pageOptions)
+    }
+
+    "render the TA (compliance) 'already confirmed' page with an authenticated user if consignment status is 'Completed'" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val controller: TransferAgreementController2 = taHelper.instantiateTransferAgreement2Controller(getAuthorisedSecurityComponents, app.configuration)
+      taHelper.mockGetConsignmentStatusGraphqlResponse(app.configuration, Some("Completed"))
+
+      val transferAgreementPage = controller.transferAgreement(consignmentId)
+        .apply(FakeRequest(GET, "/consignment/c2efd3e6-6664-4582-8c28-dcf891f60e68/transfer-agreement2").withCSRFToken)
+      val transferAgreementPageAsString = contentAsString(transferAgreementPage)
+
+      playStatus(transferAgreementPage) mustBe OK
+      contentType(transferAgreementPage) mustBe Some("text/html")
+      headers(transferAgreementPage) mustBe TreeMap("Cache-Control" -> "no-store, must-revalidate")
+      transferAgreementPageAsString must include(
+        s"""href="/consignment/c2efd3e6-6664-4582-8c28-dcf891f60e68/upload">
+           |                Continue""".stripMargin)
+      transferAgreementPageAsString must include("You have already confirmed all statements")
+      taHelper.checkHtmlOfComplianceFormOptions.checkForOptionAndItsAttributes(transferAgreementPageAsString, formSuccessfullySubmitted = true)
+    }
+
+    "render the TA (compliance) 'already confirmed' page with an authenticated user if user navigates back to TA page" +
+      "after successfully submitting TA form that had been incorrectly submitted (empty) prior" in {
```

Comment:
But this test is for when you've submitted a correct (fully filled in) form *after* previously submitting an incorrect (empty) one. What you've wrote sounds like the user has successfully submitted an empty form; unless I'm mistaken?

The reason for this test was that before when you submitted and incomplete form (which shows you the TA page with your mistakes), then submitted it correctly, when you went back to the page from the upload page, it wouldn't display the "Already confirmed" page, instead it would display the TA page with the errors you encountered on the first try.

Reply:
Ah, OK, fair enough. Maybe reword it to: `... after successfully submiting TA form after previously sumbitting and empty form` What do you think? It's quite a hard description to write.